### PR TITLE
fix: TT-140 time entry info is lost if project is selected at last

### PR DESCRIPTION
**Pull request by LEON12699**
It is necessary to choose project to enter the data

### CODE REVIEW

**Review by scastillo-jp — APPROVED** [review]

**Comment by scastillo-jp** [comment]
@here this change looks good,  lost if project is selected at last,  good job @Israel León Gálvez  @William Bravo TT-140 time entry info is 
Just one change adds a border radius at the input for the entry / exit time. so that everything looks similar.
Screen Shot 2021-02-05 at 11 45 54 AM

File: src/app/modules/shared/components/details-fields/details-fields.component.html [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
         formControlName="project_name"
         [data]="listProjects"
         [searchKeyword]="keyword"
         historyIdentifier="projectsSelected"
         notFoundText="No projects found"
         placeHolder="Enter the project name"
         [itemTemplate]="itemTemplate"
-        [notFoundTemplate]="notFoundTemplate">
+        [notFoundTemplate]="notFoundTemplate"
+        [class.valid]="project_name.invalid"
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.html:26]: I know I said to add a feedback. But looking at the UI, It doesn't look ok for me at first. Let's remove this line `[class.valid]="project_name.invalid"` and think another way of feedback.

image

File: src/app/modules/shared/components/details-fields/details-fields.component.html [hunk 2]
```diff
@@ -34,108 +43,112 @@
...
-      >
-        <option value="" selected="selected"></option>
-        <option *ngFor="let activity of activities" value="{{ activity.id }}">{{ activity.name }}</option>
-      </select>
+  <fieldset [disabled]="!(project_id.value && project_name.value)">
+    <div class="form-group row">
+      <label class="col-12 col-sm-2 col-form-label">Activity:</label>
+      <div class="col-12 col-sm-10">
+        <select
+          [class.is-invalid]="activity_id.invalid"
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.html:51]: I know this is not part of your ticket. But let's improve the activity field feedback by changing this line to `[class.is-invalid]="activity_id.invalid && activity_id.touched"`

File: src/app/modules/shared/components/details-fields/details-fields.component.scss [hunk 3]
```diff
@@ -93,3 +93,9 @@ input[type="date"]::-webkit-clear-button {
   border: none;
   margin: 0em;
 }
+
+.valid ::ng-deep .autocomplete-container ::ng-deep .input-container ::ng-deep input  {
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.scss:97]: What would be the purpose of this new style?

**Review by Angeluz-07 — CHANGES_REQUESTED** [review]

File: src/app/modules/shared/components/details-fields/details-fields.component.html (hunk 1 above)
Comment (Angeluz-07) [reply src/app/modules/shared/components/details-fields/details-fields.component.html:26]: We can also discuss it with the team, to get a consensus.

File: src/app/modules/shared/components/details-fields/details-fields.component.scss (hunk 3 above)
Comment (LEON12699) [reply src/app/modules/shared/components/details-fields/details-fields.component.scss:97]: It is the only way to change the style of the "ngx-autocomplete" component without affecting the default style of the component. It's use only for border-color of the input

File: src/app/modules/shared/components/details-fields/details-fields.component.ts [hunk 4]
```diff
@@ -246,6 +246,12 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
     this.saveEntry.emit({ entry, shouldRestartEntry: this.shouldRestartEntry });
   }
 
+  onclickFormAction(isProjectSelected: boolean){
+    if (isProjectSelected){
+      this.toastrService.warning('Please, select first a project');
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.ts:251]: I think we could change the message to "Please, first select a project"

Comment (LEON12699) [reply src/app/modules/shared/components/details-fields/details-fields.component.ts:251]: Ok i´m going to  change it

**Review by Angeluz-07 — APPROVED** [review]
Well done.

**Review by scastillo-jp — APPROVED** [review]

**Pull request merged**